### DISCUSSION TITLE
question: why no autofix for `require-error` and `go-require` ?

### DISCUSSION THREAD

**Discussion by ccoVeille** [Antonboom/testifylint#147] [Q&A] [answered]
These are two separate questions:

- why no autofix for `require-error` ?

I mean if a code is using `assert.NoError` or `assert.Error`, we could autofix it no ?

I looked at the edge case, like http handler/subroutine, but I didn't find a reason to do not apply it.

Maybe it's simply something I'm missing, and then it would be interesting to document it. Or it is something that was impossible to handle at first, but that could be activated now.

- why no autofix for `go-require` ?

Maybe it's simply something I'm missing, and then it would be interesting to document it.

**Comment by Antonboom** [comment, marked as answer]
`require-error` requires a lot of context around. There are many patterns with conditional asserts, group of asserts, etc. But to be honest these patterns are ignored and we can fix warned asserts. The single issue here assert objects – you cannot fix code like 

```go
assrtObj := assert.New(t)
assrtObj.NoError(t, err)
``` 
because there is no `reqObj` instead.

For `go-require` we have not common pattern for fixing. In most cases you need to rewrite test at all and make goroutine free from any assertions.

**Comment by ccoVeille** [comment]
Thanks for the reply.

OK, for go-require. Could you add a comment about this with the AutoFix: false in the readme?

For require-error, I assumed it was possible, thanks for confirming it.

The assrtObj edge case could be reported without adding the needed information for allowing a fix. It's already the case, right?

Having autofix for require-error would help a lot on large codebase

**Reply by Antonboom** [reply]
https://github.com/Antonboom/testifylint/issues/156

**Reply by ccoVeille** [reply]
Thanks for the partial fix issue with require-error !

About the fact the Readme.md could be improved to explain go-require often requires to rewrite the test.

Do you want me to open an issue to address it, or directly a PR? Or would you take it as it's a documentation thing?

**Reply by Antonboom** [reply]
This is already explained, isn't it? ...

**Reply by ccoVeille** [reply]
I'm so f… tired… I need holidays.

Yes, it does and in very detailed way.

Once again, I please apologize me.
